.github/workflows/rust.yml

```
@@ -19,17 +19,43 @@ jobs:
         uses: actions/cache@v3
         with:
           path: circuits/
-          key: cache-circuits-${{ hashFiles('**/out') }}
+          key: cache-circuits-${{ hashFiles('**/*.zok') }}
       - name: Compile circuit
         if: steps.cache-circuits.outputs.cache-hit != 'true'
-        run: cd circuits && zokrates compile -i fusion.zok
+        run: cd circuits && make
       - name: Compute trusted setup
         if: steps.cache-circuits.outputs.cache-hit != 'true'
-        run: cd circuits && zokrates setup
+        run: cd circuits && make setup
+      - name: Generate Solidity verifier
+        if: steps.cache-circuits.outputs.cache-hit != 'true'
+        run: cd circuits && make verifier
 
-  build:
+  build_contract:
     needs: build_circuit
     runs-on: self-hosted
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          submodules: recursive
+      - name: ⚡ Cache Contracts
+        id: cache-contracts
+        uses: actions/cache@v3
+        with:
+          path: l1-verifier
```

```suggestion
          path: l1-verifier/
```

:man_shrugging: maybe